#132.java

```diff
@@ -0,0 +1,48 @@
+/**
+132. Palindrome Partitioning II
+
+Given a string s, partition s such that every substring of the partition is a palindrome.
+
+Return the minimum cuts needed for a palindrome partitioning of s.
+
+For example, given s = "aab",
+Return 1 since the palindrome partitioning ["aa","b"] could be produced using 1 cut.
+**/
+
+
+/**
+ create a int array minCut to cache the minimum cuts needed for a palindrome partition for string from position j to the
+ end, initialize it with maximum number of possible cuts: length of substring - 1
+ 
+ minCut[0] is the minimum cuts needed for a palindrome partition for the entire string, namely what we want to return
+ 
+ create a 2-d boolean array to cache whether string is palindrome from position j to position i 
+ 
+ for substring from position j to the end:
+ try every partition possible i:
+ if substring(j, i) is palindrome, as long as minimum cut cached is smaller than 1 + minimum cut for substring(i, end), update it
+ edge case: if i reaches to end of string, then no minimum cut is needed for substring(j, i)
```

Comment:
DP transition function: minCut[j] = Math.min(minCut[j], minCut[i] + 1) && isPalindrome[j][i] == true
where j <= i <= n - 1
The function shows that minimum cuts for string from index j to the end depends on the minimum cuts for string from index i to the end if substring from index j to i is palindrome